conans/test/functional/command/search_test.py

```diff
@@ -1076,3 +1078,193 @@ class Test(ConanFile):
+        client.run("search missing/1.0@conan/stable#revision:234234234234234234 --revisions",
+                   assert_error=True)
+        self.assertIn("ERROR: Recipe not found: 'missing/1.0@conan/stable#revision'", client.out)
+
+        # Create a package
+        conanfile = """
+from conans import ConanFile
+class Test(ConanFile):
+    pass
+"""
+        client.save({"conanfile.py": conanfile})
+        client.run("create . lib/1.0@conan/stable")
+        # Search the wrong revision
+        client.run("search lib/1.0@conan/stable#revision:234234234234234234 --revisions",
+                   assert_error=True)
+        self.assertIn("ERROR: Recipe not found: 'lib/1.0@conan/stable#revision'", client.out)
+        # Search the right revision but wrong package
+        client.run("search lib/1.0@conan/stable#bd761686d5c57b31f4cd85fd0329751f:"
+                   "234234234234234234 --revisions", assert_error=True)
+        self.assertIn("ERROR: Package not found: "
+                      "'lib/1.0@conan/stable#bd761686d5c57b31f4cd85fd0329751f:"
+                      "234234234234234234'", client.out)
+
+        # IN REMOTE
+
+        # Search not found in remote
+        client.run("search missing/1.0@conan/stable --revisions -r default", assert_error=True)
+        self.assertIn("ERROR: Recipe not found: 'missing/1.0@conan/stable'", client.out)
+
+        # Search for package revisions for a non existing recipe in remote
+        client.run("search missing/1.0@conan/stable#revision:234234234234234234 --revisions "
+                   "-r default", assert_error=True)
+        self.assertIn("ERROR: Recipe not found: 'missing/1.0@conan/stable#revision'", client.out)
+
+        # Search for a wrong revision in the remote
+        client.run("upload lib/1.0@conan/stable -c --all")
+        client.run("search lib/1.0@conan/stable#revision:234234234234234234 --revisions "
+                   "-r default", assert_error=True)
+        self.assertIn("ERROR: Recipe not found: 'lib/1.0@conan/stable#revision'", client.out)
+
+        # Search the right revision but wrong package
+        client.run("search lib/1.0@conan/stable#bd761686d5c57b31f4cd85fd0329751f:"
+                   "234234234234234234 --revisions -r default", assert_error=True)
+        self.assertIn("ERROR: Package not found: "
+                      "'lib/1.0@conan/stable#bd761686d5c57b31f4cd85fd0329751f:"
+                      "234234234234234234'", client.out)
+
+    def search_revision_fail_if_v1_server_test(self):
+        # V1 server
+        test_server = TestServer(users={"conan": "password"}, server_capabilities=[])
+        servers = {"default": test_server}
+        client = TestClient(servers=servers, users={"default": [("conan", "password")]})
+        client.run("search missing/1.0@conan/stable --revisions -r default", assert_error=True)
+        self.assertIn("ERROR: The remote doesn't support revisions", client.out)
+
+
```

Comment:
The other exceptions you have introduced are the `Cannot list the revisions of a specific revision` for recipes (and I have suggested them for packages too)